jans-cli-tui/setup.py

```
@@ -6,10 +6,24 @@
+
+def _download_with_retry(url, dest, retries=5, delay=10):
+    for attempt in range(retries):
+        try:
+            urlretrieve(url, dest)
+            return
+        except (URLError, OSError) as e:
+            if attempt < retries - 1:
+                print("Download failed ({}), retrying in {}s...".format(e, delay))
+                time.sleep(delay)
+            else:
+                raise
```

Ruff (0.15.21)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)